code/modules/mob/living/carbon/xenomorph/castes/warlock/abilities_warlock.dm

```diff
@@ -0,0 +1,580 @@
+	var/crush_cost = plasma_cost * current_iterations
+	if(crush_cost > xeno_owner.plasma_stored)
+		owner.balloon_alert(owner, "[crush_cost - xeno_owner.plasma_stored] more plasma!")
+		stop_crush(target)
+		return
+	if(!check_distance(target))
+		stop_crush(target)
+		return
+
+	succeed_activate(crush_cost)
+	playsound(target, 'sound/effects/EMPulse.ogg', 70)
+	apply_filters(target_turfs)
+	addtimer(CALLBACK(src, .proc/remove_all_filters), 1 SECONDS, TIMER_STOPPABLE)
+
+	for(var/turf/effected_turf AS in target_turfs)
+		for(var/i in effected_turf)
+			if(iscarbon(i))
+				var/mob/living/carbon/carbon_victim = i
+				if(isxeno(carbon_victim) || carbon_victim.stat == DEAD)
+					continue
+				var/block = carbon_victim.get_soft_armor(BOMB)
+				carbon_victim.apply_damage(xeno_owner.xeno_caste.crush_strength, BRUTE, blocked = block)
+				carbon_victim.apply_damage(xeno_owner.xeno_caste.crush_strength * 1.5, STAMINA, blocked = block)
+				carbon_victim.adjust_stagger(5)
+				carbon_victim.add_slowdown(6)
+			else if(ismecha(i))
+				var/obj/vehicle/sealed/mecha/mecha_victim = i
+				mecha_victim.take_damage(xeno_owner.xeno_caste.crush_strength * 5, BOMB)
+			else if(isobj(i))
+				var/obj/obj_victim = i
+				if(istype(obj_victim, /obj/alien))
+					continue
+				obj_victim.take_damage(xeno_owner.xeno_caste.crush_strength, BOMB)
+	stop_crush(target)
+
+/// stops channeling and unregisters all listeners, resetting the ability
+/datum/action/xeno_action/activable/psy_crush/proc/stop_crush(turf/target)
+	SIGNAL_HANDLER
+	var/mob/living/carbon/xenomorph/xeno_owner = owner
+	if(channel_loop_timer)
+		deltimer(channel_loop_timer)
+		channel_loop_timer = null
+	QDEL_LIST(effect_list)
+	current_iterations = 0
+	target_turfs = null
+	effect_list = null
+	owner.remove_movespeed_modifier(MOVESPEED_ID_WARLOCK_CHANNELING)
+	action_icon_state = "psy_crush"
+	xeno_owner.update_glow()
+	add_cooldown()
+	QDEL_NULL(particle_holder)
+	UnregisterSignal(owner, list(SIGNAL_ADDTRAIT(TRAIT_FLOORED), SIGNAL_ADDTRAIT(TRAIT_INCAPACITATED)))
+
+///Apply a filter on all items in the list of turfs
+/datum/action/xeno_action/activable/psy_crush/proc/apply_filters(list/turfs)
+	LAZYINITLIST(filters_applied)
+	for(var/turf/targeted AS in turfs)
+		targeted.add_filter("crushblur", 1, radial_blur_filter(0.3))
+		filters_applied += targeted
+		for(var/atom/movable/item AS in targeted.contents)
+			item.add_filter("crushblur", 1, radial_blur_filter(0.3))
+			filters_applied += item
+	GLOB.round_statistics.psy_crushes++
+	SSblackbox.record_feedback("tally", "round_statistics", 1, "psy_crushes")
+
+///Remove all filters of items in filters_applied
+/datum/action/xeno_action/activable/psy_crush/proc/remove_all_filters()
+	for(var/atom/thing AS in filters_applied)
+		if(QDELETED(thing))
+			continue
+		thing.remove_filter("crushblur")
+	filters_applied = null
+
+/datum/action/xeno_action/activable/psy_crush/on_cooldown_finish()
+	owner.balloon_alert(owner, "Crush ready")
```

Comment:
```suggestion
	owner.balloon_alert(owner, "Crush ready")
```
There's no chat message for when the cooldown is up. See the praetorian:
![image](https://user-images.githubusercontent.com/35091844/204125436-899f82be-e470-4902-9bb0-6c08245f7e7f.png)

Reply:
I agree with Tivi here that a chat message isn't needed along side the balloon alert. Ideally there'd be a audio prompt as well but I need to beg someone for more sounds to do that.